In [1]:
import pandas as pd
import numpy as np

In [3]:
weather_2022= pd.read_csv('../data/bna_2022.csv')

In [5]:
weather_2022.head()

key        class  expire_time_gmt obs_id   obs_name  valid_time_gmt  \
0  KBNA  observation       1641027180   KBNA  Nashville      1641019980   
1  KBNA  observation       1641030780   KBNA  Nashville      1641023580   
2  KBNA  observation       1641034380   KBNA  Nashville      1641027180   
3  KBNA  observation       1641037980   KBNA  Nashville      1641030780   
4  KBNA  observation       1641041580   KBNA  Nashville      1641034380   

  day_ind  temp  wx_icon  icon_extd  ... clds  water_temp primary_wave_period  \
0       N    71       26       2600  ...  OVC         NaN                 NaN   
1       N    72       26       2600  ...  OVC         NaN                 NaN   
2       N    73       26       2600  ...  OVC         NaN                 NaN   
3       N    73       26       2600  ...  OVC         NaN                 NaN   
4       N    73       26       2600  ...  OVC         NaN                 NaN   

   primary_wave_height  primary_swell_period  primary_swell_height  \
0                  NaN                   NaN                   NaN   
1                  NaN                   NaN                   NaN   
2                  NaN                   NaN                   NaN   
3                  NaN                   NaN                   NaN   
4                  NaN                   NaN                   NaN   

   primary_swell_direction  secondary_swell_period  secondary_swell_height  \
0                      NaN                     NaN                     NaN   
1                      NaN                     NaN                     NaN   
2                      NaN                     NaN                     NaN   
3                      NaN                     NaN                     NaN   
4                      NaN                     NaN                     NaN   

   secondary_swell_direction  
0                        NaN  
1                        NaN  
2                        NaN  
3                        NaN  
4                        NaN  

[5 rows x 45 columns]

In [20]:
weather = pd.read_csv('../data/bna_weather.csv')

In [21]:
weather.head()

Date   key        class  expire_time_gmt obs_id   obs_name  \
0  2021-11-01 00:53:00  KBNA  observation       1635753180   KBNA  Nashville   
1  2021-11-01 01:30:00  KBNA  observation       1635755400   KBNA  Nashville   
2  2021-11-01 01:53:00  KBNA  observation       1635756780   KBNA  Nashville   
3  2021-11-01 02:53:00  KBNA  observation       1635760380   KBNA  Nashville   
4  2021-11-01 03:53:00  KBNA  observation       1635763980   KBNA  Nashville   

   valid_time_gmt day_ind  temp  wx_icon  ...  clds water_temp  \
0      1635745980       N    51       26  ...   OVC        NaN   
1      1635748200       N    50       29  ...   SCT        NaN   
2      1635749580       N    49       33  ...   CLR        NaN   
3      1635753180       N    48       33  ...   CLR        NaN   
4      1635756780       N    47       33  ...   CLR        NaN   

   primary_wave_period primary_wave_height  primary_swell_period  \
0                  NaN                 NaN                   NaN   
1                  NaN                 NaN                   NaN   
2                  NaN                 NaN                   NaN   
3                  NaN                 NaN                   NaN   
4                  NaN                 NaN                   NaN   

   primary_swell_height  primary_swell_direction  secondary_swell_period  \
0                   NaN                      NaN                     NaN   
1                   NaN                      NaN                     NaN   
2                   NaN                      NaN                     NaN   
3                   NaN                      NaN                     NaN   
4                   NaN                      NaN                     NaN   

   secondary_swell_height  secondary_swell_direction  
0                     NaN                        NaN  
1                     NaN                        NaN  
2                     NaN                        NaN  
3                     NaN                        NaN  
4                     NaN                        NaN  

[5 rows x 46 columns]

In [22]:
headway = pd.read_csv('../data/Headway Data.csv')

In [9]:
headway.head()

CALENDAR_ID  SERVICE_ABBR  ADHERENCE_ID        DATE  ROUTE_ABBR  \
0    120211101             1      76447164  2021-11-01           7   
1    120211101             1      76447165  2021-11-01           7   
2    120211101             1      76447166  2021-11-01           7   
3    120211101             1      76447167  2021-11-01          50   
4    120211101             1      76447168  2021-11-01          50   

   BLOCK_ABBR  OPERATOR  TRIP_ID  OVERLOAD_ID ROUTE_DIRECTION_NAME  ...  \
0        1704      2088   297750            0          TO DOWNTOWN  ...   
1        1704      2088   297750            0          TO DOWNTOWN  ...   
2        1704      2088   297750            0          TO DOWNTOWN  ...   
3        1704      2088   297749            0          TO DOWNTOWN  ...   
4        1704      2088   297749            0          TO DOWNTOWN  ...   

  ACTUAL_HDWY  HDWY_DEV  ADJUSTED_EARLY_COUNT  ADJUSTED_LATE_COUNT  \
0   15.983333  1.983333                     0                    0   
1   17.333333  3.333333                     0                    0   
2         NaN       NaN                     0                    1   
3         NaN       NaN                     0                    0   
4         NaN       NaN                     0                    0   

   ADJUSTED_ONTIME_COUNT STOP_CANCELLED PREV_SCHED_STOP_CANCELLED IS_RELIEF  \
0                      1              0                       0.0         0   
1                      1              0                       0.0         0   
2                      0              0                       NaN         0   
3                      1              0                       0.0         0   
4                      1              0                       NaN         0   

   BLOCK_STOP_ORDER  DWELL_IN_MINS  
0                 2      12.650000  
1                11       0.000000  
2                23      22.416666  
3                24       5.766666  
4                25       0.000000  

[5 rows x 30 columns]

In [14]:
headway['ACTUAL_HDWY'].median()

16.183333

In [15]:
headway['ACTUAL_HDWY'].mean()

18.989718607366832

In [16]:
headway['ACTUAL_HDWY'].max()

1084.616666

In [17]:
headway['HDWY_DEV'].median()

0.0

In [18]:
headway['HDWY_DEV'].mean()

0.3647395373397392

In [19]:
headway['HDWY_DEV'].max()

1054.616666

#### Headway
actual median - 16.183    dev median - 0.0
actual mean - 18.989      dev mean - .3647
actaul max - 1084.616     dev max - 1054.61666


In [23]:
weather.isna().sum()

Date                            0
key                             0
class                           0
expire_time_gmt                 0
obs_id                          0
obs_name                        0
valid_time_gmt                  0
day_ind                         0
temp                            0
wx_icon                         0
icon_extd                       0
wx_phrase                       0
pressure_tend                6854
pressure_desc                6854
dewPt                           0
heat_index                      0
rh                              0
pressure                        0
vis                             4
wc                              8
wdir                         1522
wdir_cardinal                   8
gust                         8279
wspd                            8
max_temp                     9378
min_temp                     9378
precip_total                 9636
precip_hrly                     0
snow_hrly                    9730
uv_desc       

In [24]:
weather.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9739 entries, 0 to 9738
Data columns (total 46 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Date                       9739 non-null   object 
 1   key                        9739 non-null   object 
 2   class                      9739 non-null   object 
 3   expire_time_gmt            9739 non-null   int64  
 4   obs_id                     9739 non-null   object 
 5   obs_name                   9739 non-null   object 
 6   valid_time_gmt             9739 non-null   int64  
 7   day_ind                    9739 non-null   object 
 8   temp                       9739 non-null   int64  
 9   wx_icon                    9739 non-null   int64  
 10  icon_extd                  9739 non-null   int64  
 11  wx_phrase                  9739 non-null   object 
 12  pressure_tend              2885 non-null   float64
 13  pressure_desc              2885 non-null   objec